src/boot/bootctl-install.c

```diff
@@ -334,6 +337,9 @@ static int update_efi_boot_binaries(const char *esp_path, const char *source_pat
                 _cleanup_close_ int fd = -EBADF;
                 _cleanup_free_ char *v = NULL;
 
+                if (de->d_type != DT_REG)
```

Comment:
we can't expct d_type to always be initialized, not all file systems support that. Hence, please change to `if (!IN_SET(de->d_type, DT_REG, DT_UNKNOWN))`…

Reply:
(note that get_file_version() contains a verify_regular_file() check actually, but weirdly turns it into ESRCH, something weird is going on there)